### PR TITLE
Add high density theme

### PR DESCRIPTION
**Pull request by dancormier**
This PR adds a _high density theme_ to Stacks. It does this by scaling spacing units (`--su-[x]` variables) and line-heights (`--lh-[x] variables`) using their base variable values (`--su-base` and `--lh-base` respectively).
To that end, this PR also:
- Converts `@su[x]` Less variables to CSS variables based off of `--su-base`
- Adds `--sus[x]` variables for static spacing units _not_ impacted by high density mode
- Removes `@lh-[x]` Less variables, replacing them with CSS variables based off of `--lh-base`

<details>
<summary>Screenshots</summary>

</details>

### CODE REVIEW

**Comment by aaronshekey** [comment]
Should we only be changing margin and padding in our approach to high density? We'll be less likely to trash a layout that has defined widths, no?

**Comment by dancormier** [comment]
@aaronshekey yeah, I'm guessing that 2021 me did a _find and replace_ and height/width got `--su` when they should've got `--sus`. Commit incoming.

File: lib/css/components/_stacks-user-cards.less [hunk 1]
```diff
@@ -58,22 +55,9 @@
...
-        }
-
-        .s-user-card--link {
-            margin-right: var(--su6);
-        }
-
-        .s-user-card--avatar {
-            margin-right: var(--su6);
-        }
-
```
Comment (dancormier) [inline lib/css/components/_stacks-user-cards.less]: This stuff getting yeeted feels wrong. Might have been some merge weirdness.

**Comment by dancormier** [comment]
Closing in favor of #873

**Pull request closed without merging**